python/src/aiconfig/editor/client/src/components/SettingsPropertyRenderer.tsx

```diff
@@ -367,8 +395,14 @@ export default function SettingsPropertyRenderer({
             <Stack>{subpropertyControls}</Stack>
           </>
         );
+      } else {
+        propertyControl = (
+          <JSONEditor
+            content={initialValue as JSONObject}
+            onChangeContent={setAndPropagateValue}
+          />
```

Comment:
```suggestion
        <Stack>
             <PropertyLabel
                propertyName={propertyName}
                propertyDescription={propertyDescription}
              />
            <JSONEditor
            content={initialValue as JSONObject}
            onChangeContent={setAndPropagateValue}
          />
        </Stack>
```

Let's add the label

Reply:
Also can you play around with it to see if we can get the width to look nicer? Maybe wrap the editor here with a div with 100% width or something?

Reply:
added label. Tried setting width to 100%, that did nothing. Anything > 100 would make it render nice but it would throw an exception on render

Reply:
Ya we won't want to go over 100% because there will be logic happening in monaco editor internals to handle resizing, probably based on parent element or something like that, and exceeding the parent width will cause troubles there